feature-libs/asm/components/customer-emulation/customer-emulation.component.html

```diff
@@ -1,11 +1,73 @@
 <ng-container *ngIf="isCustomerEmulationSessionInProgress$ | async">
-  <input
-    formcontrolname="customer"
-    type="text"
-    disabled="true"
-    placeholder="{{ customer?.name }}, {{ customer?.uid }}"
-  />
-  <button (click)="logoutCustomer()">
+  <div class="customerInfo">
+    <label class="name">{{ customer?.name }}</label>
+    <label class="uid">{{ customer?.uid }}</label>
+  </div>
+  <label for="cartNumber"
+    >{{ 'asm.assignCart.cartNumber' | cxTranslate }}
+    <div
+      role="search"
+      aria-label="Assign a cart id to customer"
+      class="assignCart"
+      [class.active]="cartIdExists"
+      (click)="cartIdElement.focus()"
+    >
+      <input
+        autocomplete="off"
+        placeholder=""
+        #cartIdElement
+        formcontrolname="cartNumber"
+        [formControl]="cartId"
+        (keydown.enter)="assignCartToCustomer()"
+        aria-label="Enter cart id"
+      />
+      <button
+        class="reset"
+        aria-label="Reset"
+        [class.visible]="cartIdExists"
+        (click)="clearText()"
+      >
+        <cx-icon
+          class="cx-icon fas fa-times-circle"
+          ng-reflect-type="RESET"
+        ></cx-icon>
+      </button>
+      <div role="presentation" class="search-icon">
+        <cx-icon
+          class="cx-icon fas fa-search"
+          ng-reflect-type="SEARCH"
+        ></cx-icon>
+      </div>
+      <label class="message success" [class.show]="showAssignCartSuccess">
+        <cx-icon
+          class="cx-icon fas fa-check-circle"
+          ng-reflect-type="SUCCESS"
+        ></cx-icon>
+        {{ 'asm.assignCart.success' | cxTranslate }}</label
+      >
+      <label class="message error" [class.show]="showAssignCartError">
+        <cx-icon
+          class="cx-icon fas fa-exclamation-circle"
+          ng-reflect-type="ERROR"
+        ></cx-icon>
+        {{ 'asm.assignCart.error' | cxTranslate }}</label
+      >
+    </div>
+    <button
+      class="assignCartToCustomer"
+      [disabled]="!cartIdExists"
+      (click)="assignCartToCustomer()"
+      [class.active]="cartIdExists"
+      data-cy="assignCart"
```

Comment:
About `data-cy`:  The storefront markup should be free of any test-related attributes.  Please remove `data-cy` and similar attributes for test usage only.
Instead, we can use simple css selectors.
So far, it was possible to select the right element in unit tests and e2es using this technique.